packages/notification-services-controller/src/NotificationServicesController/NotificationServicesController.test.ts

```diff
@@ -477,68 +488,87 @@ describe('metamask-notifications - fetchAndUpdateMetamaskNotifications()', () =>
     };
   };
 
-  it('processes and shows feature announcements, wallet and snap notifications', async () => {
-    const {
-      messenger,
-      mockFeatureAnnouncementAPIResult,
-      mockListNotificationsAPIResult,
-    } = arrangeMocks();
-
-    const snapNotification = createMockSnapNotification();
-    const processedSnapNotification = processSnapNotification(snapNotification);
-
+  const arrangeController = (
+    messenger: NotificationServicesControllerMessenger,
+    overrideState?: Partial<NotificationServicesControllerState>,
+  ) => {
     const controller = new NotificationServicesController({
       messenger,
       env: { featureAnnouncements: featureAnnouncementsEnv },
       state: {
         ...defaultState,
+        isNotificationServicesEnabled: true,
         isFeatureAnnouncementsEnabled: true,
-        metamaskNotificationsList: [processedSnapNotification],
+        ...overrideState,
       },
     });
 
+    return controller;
+  };
+
+  it('processes and shows all notifications (announcements, wallet, and snap notifications)', async () => {
```

Comment:
The tests for the `fetchAndUpdateMetamaskNotifications` method were messy, cleaned them up.